### ISSUE TITLE
Versioning Acorn Apps

### ISSUE THREAD

**Issue by josedev-union** [acorn-io/acorn#722]
It would be good to have a new section in Acornfile or a dedicated file to include app metadata such as version.
There is no way to navigate the acorn source of the exact version.
Considering the fact that acorn app is packaged in OCI image format, versioning acorn app should enforce the tag in image build and push, I think.

**Comment by josedev-union** [comment]
> I don't understand what you're asking for here. It's unclear to me what acorn should be exposing or enforcing. Can you give a more concrete example?

Users can specify any tag when build acorn image `acorn build -t`. it means acorn app version and image version are different.
So I think acorn image build should use the tag infor or at least version infor from the acorn app metadata.

**Comment by cloudnautique** [comment]
@josedev-union are you asking for Acorn to support labels on the Acorn app itself? 

Something like the way Docker does with the [label](https://docs.docker.com/engine/reference/builder/#label)?

**Comment by josedev-union** [comment]
> @josedev-union are you asking for Acorn to support labels on the Acorn app itself?
> 
> Something like the way Docker does with the [label](https://docs.docker.com/engine/reference/builder/#label)?

Yes, I thought of image tagging to necessarily include the app version, but it doesn't seem necessary to include it in the tag. 
When Acornfile is presented in multiple files, we can consider a dedicated file for LABELs

**Comment by josedev-union** [comment]
> Acorn supports labels in the acornfile: https://docs.acorn.io/authoring/labels
> 
> Does that meet your needs?

that is enough.
Btw, what can be the best practice for versioning?
I thought about 2 options:
For example, I wrote a mongodb acornfile, and added `version` arg to specify mongodb version. So users can specify any version via arg they want and it means there is no need for acorn image tag to include mongodb version infor, and it will just involve the acornfile version, not application version. Do you think this is a good idea?
Or set mongodb version as static in acornfile and use label to record mongo version explicitly in acornfile, and then I build acorn image with a tag including mongo db version. It means we need to build acorn image per mongodb version.


Ofc, this can be up to user but I think it would be good to mention the best practice in the doc so developers can follow

**Comment by cjellick** [comment]
Best practice would be an acorn image per mongodb version. This is the pattern we are following in our library of images:
https://github.com/acorn-io/library

I'm going to go ahead and convert this issue to a discussion.
